### ISSUE TITLE
Provide a Setup file + Button to instantly start coding

### ISSUE THREAD

**Issue by laeubi** [eclipse-m2e/m2e-core#120]
@merks has described a great opprotunity to have a quick-start fro contributors, we should offer a quick-link as this is great for new contributors and current instructions are hard to follow.

Copied from the mailing list:

> I think I recently read about that's it is possible to have a "button" in a html page that links to an Oomph setup file and simply creates the IDE+checkouts and so on. I think that should be the very first thing any project that wishes to attract developers should setup. 

Yes, this page has a link with such a button:

  https://www.eclipse.org/setups/installer/?url=https://git.eclipse.org/c/oomph/org.eclipse.oomph.git/plain/setups/interim/PlatformSDKConfiguration.setup&show=true

I.e., like this for those who don't follow links:
![grafik](https://user-images.githubusercontent.com/1331477/112962554-cb54c280-9146-11eb-93b4-14f9d9b57372.png)

The actual link URL on the button is this:
  eclipse+installer:https://git.eclipse.org/c/oomph/org.eclipse.oomph.git/plain/setups/interim/PlatformSDKConfiguration.setup

Most people use the installer in a disposable way, but of course one must keep it around more permanently and register it as a link handler to make direct use of such a single-click link.  That's why I provide a generic page through which to indirect the configuration link such that the user has documentation about different ways to use the configuration link.

The embedded documentation in the Configuration itself is also included on this indirection page.  In this case that documentation has a link to the tutorial documentation that I wrote specifically for the overall platform's configuration to make it easy for anyone to get started in order contribute to any part of the platform or all parts of the platform:

https://wiki.eclipse.org/Eclipse_Platform_SDK_Provisioning

So mostly one just needs a Project setup (I've already written ones for all the platform projects, so that's done), and a Configuration (which essentially just selects which Product Version to install and which Project Streams to provision):

https://wiki.eclipse.org/Eclipse_Oomph_Authoring#Automation_and_Specialization_with_Configurations

**Comment by mickaelistria** [comment]
That would be a good addition to CONTRIBUTING.md , before the "Clone in Eclipse" button.
The issue you'll see is that GitHub doesn't allow links to non http/https URLs. So you'll probably need to build a redirect service for that, similarly to what's done in https://github.com/mickaelistria/redirctToEclipseIDECloneCommand . This repo could even be extended/forked to directly add the necessary magic for Oomph in it.

**Comment by laeubi** [comment]
I think it would be great to have this redirect on eclipse.org level (I even think a special domain like https://install.eclipse.org would be cool) so we can have https://install.eclipse.org?setup=<the target> even more cool would be if that service would access the Oompf catalogue so we can even have https://install.eclipse.org?product=m2e-dev

The page could then display instructions how to get the Eclipse installer + information about the product and so on... It could use a new frame for actually triggering the install if already installed and so on...

**Comment by merks** [comment]
Note that we already have a reusable http/https link you can use.  E.g., it's used like this as well:

https://www.eclipse.org/setups/installer/?url=http://git.eclipse.org/c/emf/org.eclipse.emf.git/plain/releng/org.eclipse.emf.releng/EMFDevelopmentEnvironmentConfiguration.setup&show=true
https://www.eclipse.org/setups/installer/?url=http://git.eclipse.org/c/oomph/org.eclipse.oomph.git/plain/setups/configurations/OomphConfiguration.setup&show=true

So all you need is the Project setup and Configuration, including in the Configuration the documentation you want to see in the header section to customize the page contents...

**Comment by merks** [comment]
If/when you need help, I'll watch here...

**Comment by HannesWell** [comment]
Is anyone already working on the setup file? If not I offer to create one.

**Comment by merks** [comment]
I can help review what you create (with limited available time).

**Comment by HannesWell** [comment]
I've just opened a PR adding an initial Oomph setup.

**Comment by HannesWell** [comment]
To ask for inclusion of the created Oomph-Setup into the Eclipse.org project catalog I submitted a Bugzilla against Oomph:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=573588

**Comment by HannesWell** [comment]
> To ask for inclusion of the created Oomph-Setup into the Eclipse.org project catalog I submitted a Bugzilla against Oomph:
> https://bugs.eclipse.org/bugs/show_bug.cgi?id=573588

The m2e setup is now included in Oomph's Eclipse projects catalogue. Thanks to Ed Merks.

Testers are welcome.

I've already tested the setup in a clean environment and it worked quiet well. For some minor improvements I found during that test I  opened PR #201.

In the [bugzilla](https://bugs.eclipse.org/bugs/show_bug.cgi?id=573588#c3), as well as earlier in this discussion, Ed Merks suggested to add a Configuration setup which allows to create the wanted installer auto-launch button.
If you agree I can create the configuration and add the corresponding link to the `CONTRIBUTION.md` and `README.md`.

**Comment by HannesWell** [comment]
Additionally we could extend the setup by setting some of the basic-preferences like the workspace's file encoding or more advanced once like save-action on clean-up.
Even tough personally I really like save-actions, it should be considered that it affects the entire workspace and if somebody has other projects in the workspace too, this could cause unintended changes. But if you say that users should only have the m2e-projects in their workspace it would be fine. Furthermore the code-base should be clean-up/formatted in general and this would have need to be done once. But if this is done, Save-actions are a good way to keep the code base formatted and clean.

**Comment by merks** [comment]
Note that many (most) settings can be made project-specific, but those are a lot of work to maintain consistently...

**Comment by HannesWell** [comment]
> Note that many (most) settings can be made project-specific, but those are a lot of work to maintain consistently...

Yes that's what I've seen in many Eclipse projects. But the maintenance you mentioned holds me of to use that approach in may projects.
I wonder if it is possible to refer from a preference-file of a project to another file that holds the actual key-value pairs? This way the relevant preferences could be configured in one file shared by multiple projects. Of course the approach would likely have some difficulties, e.g. how is the reference specified? By relative file-system paths or by symbolic links? But then if the referenced file is moved the preferences are broken or the referenced file has to be in the workspace. And probably a lot more. But it would reduce the effort for maintenance a lot.
But I've never seen something like this, so I think this is not possible.

**Comment by HannesWell** [comment]
The PR #201 now also adds a configuration setup that enables the use of the auto-launch installer button and adds such buttons to the README.md and CONTRIBUTING.md, which is the initial intend of this issue.

**Comment by merks** [comment]
In Oomph I have prototyped a framework for managing project-specific preferences, but it's been many years since I worked on it and it never got to production quality.  I wanted to use it for EMF...  The implementation is in the /org.eclipse.oomph.preferences* projects.   The idea is based on identifying a project that is the "profile" for a set of project-specific preference and defining how changes to that project's project-specific preferences are automatically transferred to other projects.  With this approach one would only ever have to edit one project's preferences...  But I think it's a lot of work to get this to production quality, though it would be cool and useful I think...
